cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -74,6 +68,80 @@
 _VARIADIC_GATE_FAMILIES = [_MEASUREMENT_GATE_FAMILY, _WAIT_GATE_FAMILY]
 
 
+GateOrFamily = Union[Type[cirq.Gate], cirq.Gate, cirq.GateFamily]
+
+
+@dataclass
+class _GateRepresentations:
+    """Contains equivalent representations of a gate in both DeviceSpecification and GridDevice."""
+
+    """The name of gate type in `GateSpecification`."""
+    gate_spec_name: str
+    """Gate representations to be included in generated gatesets and gate durations."""
+    primary_forms: List[GateOrFamily]
+    """GateFamilies which match all other valid gate representations."""
+    additional_forms: List[cirq.GateFamily] = field(default_factory=list)
+    """`primary_forms` (as GateFamilies) + `additional_forms`"""
+    all_forms: List[cirq.GateFamily] = field(init=False)
```

Comment:
Optional:  Should this be a generated @property instead?  I guess, with init=False, they can't specify it, so maybe that's okay.